src/model/services/mapi/capiv1alpha3/getClusterInfraRef.ts

```diff
@@ -0,0 +1,55 @@
+import { IHttpClient } from 'model/clients/HttpClient';
+import { LoggedInUserTypes } from 'stores/main/types';
+
+import * as capzv1alpha3 from '../capzv1alpha3';
+import { ICluster } from './';
+
+export function getClusterInfraRef(
+  client: IHttpClient,
+  user: ILoggedInUser,
+  cluster: ICluster
+) {
+  return async () => {
+    const { infrastructureRef } = cluster.spec;
+    if (!infrastructureRef) {
+      return Promise.reject(
+        new Error('There is no infrastructure reference defined.')
+      );
+    }
+
+    switch (infrastructureRef.kind) {
+      case capzv1alpha3.AzureCluster:
+        return capzv1alpha3.getAzureClusterByName(
+          client,
+          user,
+          cluster.metadata.namespace,
+          infrastructureRef.name
+        )();
+
```

Comment:
Just add another case for handling a different type, same as in the key function below